smolyay/basis.py

```diff
@@ -328,141 +296,151 @@ def derivative(self, x):
-        return NestedBasisFunctionSet(points,basis_functions,levels)
-
-
-class BasisFunctionSet():
-    """Set of basis functions and sample points.
+    .. math::
+
+        \sigma(n) = \begin{cases}
+            (1 + n) / 2 & \text{ if } n \text{ is odd}\\ 
+            -n / 2 & \text{ if } n \text{ is even} 
+            \end{cases}
 
     Parameters
     ----------
-    basis_functions : list
-        Basis functions in set.
-
-    points : list
-        Sample point corresponding to each basis function.
+    n : int
+        Degree of trigonometric polynomial.
 
     """
 
-    def __init__(self,points,basis_functions):
-        self._basis_functions = basis_functions
-        self._points = points
-        if len(basis_functions) != len(points):
-            raise IndexError("basis_functions and points must have the "
-                    "same number of elements.")
+    def __init__(self, n):
+        super().__init__()
+        self._n = int(n)
 
     @property
-    def points(self):
-        """list: Sampling points."""
-        return self._points
+    def domain(self):
+        """numpy.ndarray: Domain the sample points come from."""
+        return numpy.array([0, 2 * numpy.pi])
 
     @property
-    def basis_functions(self):
-        """list: Basis functions."""
-        return self._basis_functions
+    def n(self):
+        """int: Degree of polynomial."""
+        return self._n
+    
+    @n.setter
+    def n(self,value):
+        self._n = int(value)
 
+    def _function(self, x):
+        r"""Evaluate the basis function.
 
-class NestedBasisFunctionSet(BasisFunctionSet):
-    """Nested set of basis functions and points.
+        The Trigonometric polynomial is evaluated using the following formula:
 
-    Nested points/basis function grow in levels, such that an approximation
-    of a given level uses not only its sampling points but also all the points
-    at lower levels. Nested sets (similarly to ogres) are like onions.
+        .. math::
 
-    Parameters
-    ---------
-    levels : list of lists
-       Assignment of points/basis functions to each level.
-    """
+            \phi_n(x) = \exp(x i \sigma(n))
 
-    def __init__(self,points,basis_functions,levels):
-        super().__init__(points,basis_functions)
-        self._levels = levels
+        where *n* is the order of trigonometric polynomial and
+        is a non-negative integer, and `sigma` the frequency of the
+        complex exponential
+
+        Parameters
+        ----------
+        x : float
+            One-dimensional point on :math:`[0, 2\pi]`.
+
+        Returns
+        -------
+        float
+            Value of Trigonometric polynomial.
+
+        Raises
+        ------
+        ValueError
+            If input is outside the domain `[0, 2\pi]`
+        """
+        x = numpy.asarray(x)
+        if self.n % 2 == 1:
+            return numpy.exp(x * (1 + self.n) / 2 * 1j)
+        else:
+            return numpy.exp(x * -self.n / 2 * 1j)
```

Comment:
I think there may have been a miscommunication before. This should just be `n` and `n` is allowed to be negative:
```suggestion
        return numpy.exp(x * self.n  * 1j)
```